core/src/PayPal/Order/Action/CaptureAuthorizationAction.php

```diff
@@ -0,0 +1,148 @@
+                sprintf('PayPal Order %s does not have a valid authorization', $payPalOrder->getId()),
+                PsCheckoutException::PAYPAL_AUTHORIZATION_NOT_FOUND
+            );
+        }
+
+        $authorizationId = $authorization['id'];
+        $authorizationStatus = $authorization['status'];
+
+        // Check if status is VOIDED
+        if ($authorizationStatus === PayPalAuthorizationStatus::VOIDED) {
+            throw new PsCheckoutException(
+                sprintf('Authorization %s is voided and cannot be captured', $authorizationId),
+                PsCheckoutException::PAYPAL_AUTHORIZATION_VOIDED
+            );
+        }
+
+        // Validate authorization status must be CREATED or PARTIALLY_CAPTURED
+        if (!in_array($authorizationStatus, [PayPalAuthorizationStatus::CREATED, PayPalAuthorizationStatus::PARTIALLY_CAPTURED], true)) {
+            throw new PsCheckoutException(
+                sprintf(
+                    'Authorization %s status must be CREATED or PARTIALLY_CAPTURED, current status: %s',
+                    $authorizationId,
+                    $authorizationStatus
+                ),
+                PsCheckoutException::PAYPAL_AUTHORIZATION_STATUS_INVALID
+            );
+        }
+
+        // Check if expiration_time has passed
+        if (isset($authorization['expiration_time'])) {
+            try {
+                $expirationTime = new \DateTime($authorization['expiration_time']);
+                $currentTime = new \DateTime('now', new \DateTimeZone('UTC'));
+
+                if ($expirationTime < $currentTime) {
+                    throw new PsCheckoutException(
+                        sprintf('Authorization %s has expired', $authorizationId),
+                        PsCheckoutException::PAYPAL_AUTHORIZATION_EXPIRED
+                    );
+                }
+            } catch (PsCheckoutException $e) {
+                // Re-throw PsCheckoutException
+                throw $e;
+            } catch (\Exception $e) {
+                // If we can't parse the expiration time, log it but don't block the capture
+                // The PayPal API will reject it if it's actually expired
+            }
+        }
+
+        // Call captureAuthorization in PaymentHttpClient
+        $captureResponse = $this->paymentHttpClient->captureAuthorization($authorizationId);
+        $capturedAuthorization = json_decode($captureResponse->getBody(), true);
+
+        $authorizationEntity = $this->authorizationRepository->getById($capturedAuthorization['id']);
+
+        if ($authorizationEntity) {
+            $authorizationEntity->setStatus($capturedAuthorization['status']);
+        } else {
+            $authorizationEntity = new PayPalOrderAuthorization(
+                $capturedAuthorization['id'],
+                $payPalOrder->getId(),
+                $capturedAuthorization['status'] ?? null,
+                $capturedAuthorization['expiration_time'] ?? null,
```

Comment:
I don't think these values should be defaulted to `null`.

Reply:
Actually I think that we shouldn't create a new entity, because it should already exist in the db during authorization create. I will change so only status and update_date are updated on already existing entity